From 2c09395d987f18c333b15b3a33f4f412a3aac7ff Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Matthias=20A=C3=9Fhauer?= <mha1993@live.de>
Date: Wed, 17 Sep 2025 10:53:00 +0200
Subject: [PATCH 6/7] Mangle Absolute path

Test 25-test_verify.t tries to test some file:// URLs containing an absolute path,
but perls abs_path() creates MSYS2 absolute paths in our case, which our native
openssl doesn't understand.

Run the path through cygpath to convert it into the expected format.
---
 test/recipes/25-test_verify.t | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/test/recipes/25-test_verify.t b/test/recipes/25-test_verify.t
index 673c3d5..786a302 100644
--- a/test/recipes/25-test_verify.t
+++ b/test/recipes/25-test_verify.t
@@ -614,6 +614,10 @@ my $foo_file = "cert.pem";
 copy($rootcert, $foo_file);
 ok(vfy_root("-CAstore", $foo_file), "CAstore foo:file");
 my $abs_cert = abs_path($rootcert);
+if ($^O eq "msys") {
+    $abs_cert = `cygpath -m $abs_cert`;
+    chomp $abs_cert;
+}
 # Windows file: URIs should have a path part starting with a slash, i.e.
 # file://authority/C:/what/ever/foo.pem and file:///C:/what/ever/foo.pem
 # file://C:/what/ever/foo.pem is non-standard and may not be accepted.
